zebra-chain/src/transaction/serialize.rs

```diff
@@ -194,6 +207,35 @@ impl ZcashSerialize for Transaction {
                 writer.write_u32::<LittleEndian>(expiry_height.0)?;
                 inputs.zcash_serialize(&mut writer)?;
                 outputs.zcash_serialize(&mut writer)?;
+
+                // In V5 the serialized structure is aligned with the
+                // internal structure.
+
+                match sapling_shielded_data {
+                    None => {
+                        // Signal no sapling value balance, anchor variant,
+                        // shielded spends, shielded outputs.
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
+
+                        // Todo: we don't have test vector so we don't know
+                        // if we need extra 0s as in v4.
```

Comment:
We should block the V5 PR on having a range of test data.

But the V4 PR can just use existing blocks as test data.